pkg/controller/pravega/pravega_segmentstore.go

```diff
@@ -158,6 +154,27 @@ func makeSegmentstorePodSpec(p *api.PravegaCluster) corev1.PodSpec {
 	return podSpec
 }
 
+func MakeSegmentStoreVolumeMount(p *api.PravegaCluster) []corev1.VolumeMount {
+	if util.IsVersionBelow07(p.Spec.Version) {
+		return []corev1.VolumeMount{
+			{
+				Name:      cacheVolumeName,
+				MountPath: cacheVolumeMountPoint,
+			},
+			{
+				Name:      heapDumpName,
+				MountPath: heapDumpDir,
+			},
+		}
+	}
+	return []corev1.VolumeMount{
```

Comment:
Lets write more neat and readable code. First create a VolumeMount slice with heapdump mount in it. If util.IsVersionBelow07(p.Spec.Version) append cacheVolumeMount to the slice